src/extension/csv-with-metadata-header/src/main/java/au/org/emii/geoserver/wfs/response/CSVWithMetadataHeaderOutputFormat.java

```diff
@@ -26,51 +14,168 @@
+    public CSVWithMetadataHeaderOutputFormat(GeoServer gs, Catalog catalog) {
         super(gs, "csv-with-metadata-header");
         this.catalog = catalog;
-        this.csvOutputFormat = csvOutputFormat;
+        DateTimeZone.setDefault(DateTimeZone.UTC);
     }
 
-    @Override
-    public String getMimeType(Object value, Operation operation) {
-        return this.csvOutputFormat.getMimeType(value, operation);
-    }
-
-    @Override
-    public String getPreferredDisposition(Object value, Operation operation) {
-        return this.csvOutputFormat.getPreferredDisposition(value, operation);
-    }
-
-    @Override
-    public String getAttachmentFileName(Object value, Operation operation) {
-        return this.csvOutputFormat.getAttachmentFileName(value, operation);
+    public String getMimeType(Object value, Operation operation) throws ServiceException {
+        return "text/csv";
     }
 
-    @Override
     public String getCapabilitiesElementName() {
-        return this.csvOutputFormat.getCapabilitiesElementName();
+        return "CSV";
     }
 
     @Override
     protected void write(FeatureCollectionResponse featureCollection,
                          OutputStream output,
                          Operation getFeature) throws IOException, ServiceException {
         BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output));
-        writeMetadata(featureCollection, getMetadataFeatureName(featureCollection), w);
-        this.csvOutputFormat.write(featureCollection, output, getFeature);
+        //writeMetadata(featureCollection, getMetadataFeatureName(featureCollection), w);
+        writeData(featureCollection, output, getFeature);
+    }
+
+
+    public void writeData(FeatureCollectionResponse featureCollection, OutputStream output, Operation getFeature) throws IOException, ServiceException {
+        BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output, this.gs.getGlobal().getSettings().getCharset()));
+        FeatureCollection<?, ?> fc = (FeatureCollection)featureCollection.getFeature().get(0);
+        Object att;
+
+        SimpleFeatureType ft = (SimpleFeatureType)((FeatureCollection)fc).getSchema();
+        w.write("FID,");
+
+        for(int i = 0; i < ft.getAttributeCount(); ++i) {
+            AttributeDescriptor ad = ft.getDescriptor(i);
+            w.write(this.prepCSVField(ad.getLocalName()));
+            if (i < ft.getAttributeCount() - 1) {
+                w.write(",");
+            }
+        }
+
+        w.write("\r\n");
+        NumberFormat coordFormatter = NumberFormat.getInstance(Locale.US);
+        coordFormatter.setMaximumFractionDigits(this.getInfo().getGeoServer().getSettings().getNumDecimals());
+        coordFormatter.setGroupingUsed(false);
+        FeatureIterator i = ((FeatureCollection)fc).features();
+
+        try {
+            for(; i.hasNext(); w.write("\r\n")) {
```

Comment:
Would 

    while (i.hasNext) {
        ...
        w.write("\r\n");
    }

be more readable?